kernel/sem.c

```diff
@@ -191,32 +179,12 @@ int _impl_k_sem_take(struct k_sem *sem, s32_t timeout)
 }
 
 #ifdef CONFIG_USERSPACE
-u32_t _handler_k_sem_take(u32_t sem_ptr, u32_t timeout, u32_t arg3,
-			  u32_t arg4, u32_t arg5, u32_t arg6, void *ssf)
-{
-	_SYSCALL_ARG2;
-
-	_SYSCALL_OBJ(sem_ptr, K_OBJ_SEM, ssf);
-	return _impl_k_sem_take((struct k_sem *)sem_ptr, timeout);
-}
-
-u32_t _handler_k_sem_reset(u32_t sem_ptr, u32_t arg2, u32_t arg3,
-			   u32_t arg4, u32_t arg5, u32_t arg6, void *ssf)
+_SYSCALL_HANDLER2(k_sem_take, sem, timeout)
 {
-	_SYSCALL_ARG1;
-
-	_SYSCALL_OBJ(sem_ptr, K_OBJ_SEM, ssf);
-	_impl_k_sem_reset((struct k_sem *)sem_ptr);
-
-	return 0;
+	_SYSCALL_OBJ(sem, K_OBJ_SEM);
```

Comment:
This can also be a _SIMPLE syscall

Reply:
no, because of the additional timeout parameter